From ca229f6cad5cc72eafe26cd8ab14b531b18e0b6f Mon Sep 17 00:00:00 2001
From: xWASABI <wasabi.dev@gmail.com>
Date: Sat, 2 May 2015 04:18:00 -0500
Subject: [PATCH 08/10] Exposing Settings backgrounds for more theming options,
 while not impacted stock.

MODIFIED:
        - SETTINGS - Dashboard Card Background exposed to allow themers to theme with a .png, complex .xml, or still use the existing method with <color name="exposed_bg_dark>.
                - res/layout/dashboard_category.xml referenced background with @color/exposed_bg_dark, but changed to @drawable/dashboard_background_exposed.xml.
                - res/drawable/dashboard_background_exposed.xml Uses <layer-list> and references @color/exposed_bg_dark to enable old themes to remain intact and
                  to make sure stock is not impacted.

        - SETTINGS - Main Backgound behind Dashboard Cards exposed to allow themers to theme with a .png, complex .xml, or still use the existing method with
                     <color name="dashboard_background_color">.
                   - res/drawable/settings_main_background_exposed.xml Uses <layer-list> and references @color/dashboard_background_color to enable old themes to
                     remain intact and to make sure stock is not impacted.

        - VALUES - Color.xml was modified a little to make sure stock is not impacted.
                 - res/values/colors.xml The color used to change main dashboard background is <color name="dashboard_background_color">#ffe1e1e0</color> it was
                   modified to be #ffffff instead of #ffe1e1e0 to make sure stock is not impacted.

- KEEP NOTED
        - Old themes and Stock will not see in any difference with this commit this is just giving themers options to provide a better user experience.

- NOTABLE MENTION
        - Thank you Mike Williams with some ideas on how to make this happen.
---
 res/drawable/dashboard_background_exposed.xml     | 12 ++++++++++++
 res/drawable/settings_main_background_exposed.xml |  9 +++++++++
 res/layout/dashboard_category.xml                 |  4 ++--
 res/layout/settings_main_dashboard.xml            |  2 +-
 res/values/colors.xml                             |  2 +-
 5 files changed, 25 insertions(+), 4 deletions(-)
 create mode 100644 res/drawable/dashboard_background_exposed.xml
 create mode 100644 res/drawable/settings_main_background_exposed.xml

diff --git a/res/drawable/dashboard_background_exposed.xml b/res/drawable/dashboard_background_exposed.xml
new file mode 100644
index 0000000..c346af8
--- /dev/null
+++ b/res/drawable/dashboard_background_exposed.xml
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item>
+        <shape  android:shape="rectangle">
+
+            <solid android:color="@color/dashboard_background_color" />
+
+        </shape>
+    </item>
+</layer-list>
+
diff --git a/res/drawable/settings_main_background_exposed.xml b/res/drawable/settings_main_background_exposed.xml
new file mode 100644
index 0000000..ee6b75a
--- /dev/null
+++ b/res/drawable/settings_main_background_exposed.xml
@@ -0,0 +1,9 @@
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item>
+        <shape  android:shape="rectangle">
+
+            <solid android:color="@color/exposed_bg_dark" />
+
+        </shape>
+    </item>
+</layer-list>
diff --git a/res/layout/dashboard_category.xml b/res/layout/dashboard_category.xml
index 1ceeebf..5498225 100644
--- a/res/layout/dashboard_category.xml
+++ b/res/layout/dashboard_category.xml
@@ -11,7 +11,7 @@
      distributed under the License is distributed on an "AS IS" BASIS,
      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      See the License for the specific language governing permissions and
-     limitations under the License.
+     limitations under the License. 
 -->
 
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
@@ -21,7 +21,7 @@
         android:paddingStart="@dimen/dashboard_category_padding_start"
         android:paddingEnd="@dimen/dashboard_category_padding_end"
         android:orientation="vertical"
-        android:background="@color/dashboard_category_background_color"
+        android:background="@drawable/dashboard_background_exposed"
         android:layout_marginBottom="8dip"
         android:elevation="@dimen/dashboard_category_elevation">
 
diff --git a/res/layout/settings_main_dashboard.xml b/res/layout/settings_main_dashboard.xml
index 56f2737..591feca 100644
--- a/res/layout/settings_main_dashboard.xml
+++ b/res/layout/settings_main_dashboard.xml
@@ -21,5 +21,5 @@
              android:id="@+id/main_content"
              android:layout_height="match_parent"
              android:layout_width="match_parent"
-             android:background="@color/dashboard_background_color"
+             android:background="@drawable/settings_main_background_exposed"
              />
diff --git a/res/values/colors.xml b/res/values/colors.xml
index dd982eb..a852f79 100644
--- a/res/values/colors.xml
+++ b/res/values/colors.xml
@@ -73,7 +73,7 @@
     <color name="theme_primary_dark">#ff21272b</color>
     <color name="theme_accent">#ff009688</color>
 
-    <color name="dashboard_background_color">#ffe1e1e0</color>
+    <color name="dashboard_background_color">#ffffff</color>
     <color name="switchbar_background_color">#ff37474f</color>
     <color name="switch_accent_color">#ff7fcac3</color>
 
-- 
1.9.1

